server/src/main/java/org/apache/uniffle/server/event/AppPurgeEvent.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.server.event;
+
+import java.util.List;
+
+public class AppPurgeEvent extends PurgeEvent {
```

Comment:
What's the relation between ShuffleFlushEvent and PurgeEvent?

Reply:
AppPurgeEvent is the event of removing resources for App or App's stage.

Reply:
Should we unify the ShuffleFlushDataEvent and PurgeEvent?

Reply:
I think we dont need to do so. These events are the encapsulation for specified consumer, and we should keep it independent.

By the way, speaking of the `EVENT`, I have a idea that we could create an abstract event interface for all specific events. And to avoid thread pools anywhere, we could introduce the event dispatcher. And original event consumer could be registered to dispatcher. All in one will be better and helpful to collect metrics.

Reply:
> I think we dont need to do so. These events are the encapsulation for specified consumer, and we should keep it independent.
> 
> By the way, speaking of the `EVENT`, I have a idea that we could create an abstract event interface for all specific events. And to avoid thread pools anywhere, we could introduce the event dispatcher. And original event consumer could be registered to dispatcher. All in one will be better and helpful to collect metrics.

So we can add some `todo` comments here.

Reply:
It's OK.

Reply:
Done